internal/tiltfile/updatesettings/update_settings.go

```diff
@@ -26,20 +29,34 @@ func (e Extension) OnStart(env *starkit.Environment) error {
 	return env.AddBuiltin("update_settings", e.updateSettings)
 }
 
-func (e Extension) updateSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var maxParallelUpdates int
+func (e *Extension) updateSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if e.callPosition.IsValid() {
```

Comment:
Hmm...I'm concerned about  a world where every extension invents a brand-new pattern with special restrictions, which makes it harder for the user to know what to expect. 

can we just agree on doing something like this
https://github.com/tilt-dev/tilt/blob/master/internal/tiltfile/version/version.go#L38
as a standard pattern?

If we want to introduce a new rule where you can only call a _settings() function once, we should do that for all the _settings() functions.

Reply:
ah yeah I'd avoided doing that because it means I can't use the getter/setter pattern to protect from weird values getting in here -- but I can give it a shot